wallet/src/wallet-api/main.rs

```diff
@@ -1711,6 +1761,228 @@ mod tests {
             .unwrap();
         assert!(asset_info.verified);
     }
+
+    #[async_std::test]
+    #[traced_test]
+    async fn test_export_import_asset() {
+        let server = TestServer::new().await;
+        server
+            .get::<()>(&format!(
+                "newwallet/{}/password1/name/wallet1",
+                server.get::<String>("getmnemonic").await.unwrap(),
+            ))
+            .await
+            .unwrap();
+
+        let mut asset = server
+            .get::<AssetInfo>(&format!(
+                "newasset/description/{}",
+                base64::encode_config("description".as_bytes(), base64::URL_SAFE_NO_PAD)
+            ))
+            .await
+            .unwrap();
+        // We know the mint info since we created the asset. Later we will check that importers
+        // can't learn the mint info.
+        assert!(asset.mint_info.is_some());
+
+        // Export the asset.
+        let export = server
+            .get::<String>(&format!("exportasset/{}", asset.definition.code))
+            .await
+            .unwrap();
+
+        // Open a different wallet and import the asset.
+        server
+            .get::<()>(&format!(
+                "newwallet/{}/password2/name/wallet2",
+                server.get::<String>("getmnemonic").await.unwrap(),
+            ))
+            .await
+            .unwrap();
+        // Make sure the new wallet doesn't have the asset before we import it.
+        let info = server.get::<WalletSummary>("getinfo").await.unwrap();
+        assert_eq!(info.assets, vec![AssetInfo::native()]);
+
+        // Import the asset.
+        let import = server
+            .get::<AssetInfo>(&format!("importasset/{}", export))
+            .await
+            .unwrap();
+        // Make sure we didn't export the mint info.
+        assert!(import.mint_info.is_none());
+        // Check that all the information besides the mint info is the same.
+        asset.mint_info = None;
+        assert_eq!(asset, import);
+    }
+
+    #[async_std::test]
+    #[traced_test]
+    async fn test_updateasset() {
```

Comment:
If it's not too much work, it would be nice to also check that we can't update a verified asset (or is it tested in somewhere else?). This test currently only uses a native asset.

Reply:
I think this is tested pretty thoroughly in Seahorse, so it's okay if this test just focuses on the API specific stuff like route parsing and serialization.